panic on sql_table

```d2
test {
    shape: sql_table
    test_id: varchar(64) {constraint: [primary_key, foreign_key]}
}
```

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x100e364dc]

goroutine 51 [running]:
oss.terrastruct.com/d2/d2compiler.(*compiler).compileSQLTable(0x140000bea20?, 0x14001ea81c0)
        /Users/ec2-user/src/d2/d2compiler/compile.go:669 +0x70c
oss.terrastruct.com/d2/d2compiler.(*compiler).compileShapes(0x140000261f8?, 0x14001ea8000)
        /Users/ec2-user/src/d2/d2compiler/compile.go:566 +0x168
oss.terrastruct.com/d2/d2compiler.compileAST({0x0, 0x0}, {0x0?, {0x0?, 0x14000334900?, 0x1400007a720?}}, 0x140000be900)
        /Users/ec2-user/src/d2/d2compiler/compile.go:57 +0x178
oss.terrastruct.com/d2/d2compiler.Compile({0x0, 0x0}, {0x103221840, 0x1400007a720}, 0x14000334900?)
        /Users/ec2-user/src/d2/d2compiler/compile.go:40 +0xcc
oss.terrastruct.com/d2.Compile({0x1032252b8, 0x14000626040}, {0x14000034540, 0x61}, 0x14000608d78?)
        /Users/ec2-user/src/d2/d2.go:30 +0xbc
main.compile({0x1032252b8, 0x14000626040}, 0x14000608d88?, {0x103224f98, 0x1036d1e00}, 0x0, {0x16f4176ae?, 0x0?}, {0x16f4176b4, 0x6}, ...)
        /Users/ec2-user/src/d2/cmd/d2/main.go:190 +0xe0
main.(*watcher).compileLoop(0x14000634000, {0x1032252b8, 0x14000626040})
        /Users/ec2-user/src/d2/cmd/d2/watch.go:348 +0x2c4
main.(*watcher).goFunc.func1()
        /Users/ec2-user/src/d2/cmd/d2/watch.go:187 +0x7c
created by main.(*watcher).goFunc
        /Users/ec2-user/src/d2/cmd/d2/watch.go:183 +0x8c
```

:)

--- a/ci/release/changelogs/next.md
+++ b/ci/release/changelogs/next.md
@@ -19,3 +19,4 @@
 - Icons with query parameters are now being escaped to valid SVG XML. [#438](https://github.com/terrastruct/d2/issues/438)
 - Fixed connections being clipped if they were at the very top or left edges of the diagram. [#493](https://github.com/terrastruct/d2/pull/493)
 - Fixed edge case where style being defined in same scope as sql_table caused compiler to skip compiling sql_table. [#506](https://github.com/terrastruct/d2/issues/506)
+- Fixed panic when `constraint` value was not a string value. [#248](https://github.com/terrastruct/d2/issues/248)

--- a/d2compiler/compile.go
+++ b/d2compiler/compile.go
@@ -688,6 +688,10 @@ func (c *compiler) compileSQLTable(obj *d2graph.Object) {
 					continue
 				}
 				if n.MapKey.Key.Path[0].Unbox().ScalarString() == "constraint" {
+					if n.MapKey.Value.StringBox().Unbox() == nil {
+						c.errorf(n.MapKey.GetRange().Start, n.MapKey.GetRange().End, "constraint value must be a string")
+						return
+					}
 					d2Col.Constraint = n.MapKey.Value.StringBox().Unbox().ScalarString()
 				}
 			}

--- a/d2compiler/compile_test.go
+++ b/d2compiler/compile_test.go
@@ -1668,6 +1668,16 @@ choo: {
 				tassert.Equal(t, 3, len(g.Objects))
 			},
 		},
+		{
+			name: "sql-panic",
+			text: `test {
+    shape: sql_table
+    test_id: varchar(64) {constraint: [primary_key, foreign_key]}
+}
+`,
+			expErr: `d2/testdata/d2compiler/TestCompile/sql-panic.d2:3:27: constraint value must be a string
+`,
+		},
 	}
 
 	for _, tc := range testCases {

--- a/testdata/d2compiler/TestCompile/sql-panic.exp.json
+++ b/testdata/d2compiler/TestCompile/sql-panic.exp.json
@@ -1,0 +1,12 @@
+{
+  "graph": null,
+  "err": {
+    "ioerr": null,
+    "errs": [
+      {
+        "range": "d2/testdata/d2compiler/TestCompile/sql-panic.d2,2:26:54-2:64:92",
+        "errmsg": "d2/testdata/d2compiler/TestCompile/sql-panic.d2:3:27: constraint value must be a string"
+      }
+    ]
+  }
+}

